docs/rfc/rfc-2.adoc

```diff
@@ -0,0 +1,811 @@
+
+=== Current Functionality
+
+ECDSA clients currently monitor the chain for notifications that a Keep has
+closed abnormally, in particular in the function
+https://github.com/keep-network/keep-ecdsa/blob/b916eaa491b886fb6c7cde2706d225b5b85ac2a2/pkg/client/client.go#L778[`monitorKeepTerminatedEvent`].
+For tBTC, a termination indicates that the keep backing a deposit has been
+closed without a successful redemption, meaning that the keep still holds
+BTC. Upon receiving a termination event, the client waits for a set number of
+block confirmations to occur in order to be sure the event will not be
+reorged away, then archives its key material, removes the keep from the local
+list of active keeps, and finally stops monitoring for any other events
+related to that keep.
+
+Notably absent in the current ECDSA client's feature set is any interaction
+with the Bitcoin chain. In particular, the ECDSA client does not contain any
+functionality for connecting to a Bitcoin node, nor does it at any point
+assemble Bitcoin data.
+
+== Proposal
+
+=== Goal
+
+This proposal is focused on achieving a "least effort" implementation of a
+recovery protocol that can handle happy paths for BTC distribution. Where
+relevant, it points out _unhappy_ paths and may outline how they might be
+dealt with in the future, but these unhappy paths are explicitly considered
+out of scope for this RFC.
+
+Additionally, this proposal avoids _strictly requiring_ a connection to a
+Bitcoin node, as that is an additional piece of infrastructure that triggers
+additional requirements for larger staking providers that may be operating
+many nodes on behalf of multiple stakers.
+
+=== Implementation Behaviors
+
+It is important to observe that interaction with the Bitcoin chain is
+_strictly unnecessary_ to construct and sign a Bitcoin transaction that
+transmits the BTC held by a keep to n addresses, split evenly.
+
+==== Bitcoin Transaction Requirements
+
+The specific goal to recover BTC from a terminated keep is an N-way split of
+the BTC, ~1/N going to each of the N signers in the signing group. To do this, a
+Bitcoin transaction can be constructed with a single input (the UTXO that funded
+the tBTC deposit) and N outputs (one for each signer). A fee, F, is subtracted
+from the deposit's lot size, L, and (L-F)/N BTC is sent to each signer.
+
+To compute the per-signer amount, the client needs to know:
+
+- N, the number of signers in the group.
+- F, the desired fee.
+- L, the deposit lot size.
+
+Separately from the amount, the Bitcoin sighash must be computed in order to
+construct a transaction that can spend the tBTC deposit's UTXO; see the tBTC
+spec's section on https://docs.keep.network/tbtc/#sighash[Standardized
+Sighash Construction] for an introduction to this process. Computing the sighash
+requires two additional pieces of data:
+
+- The previous outpoint, which is the concatenation of the funding UTXO's
```

Comment:
Is outpoint a keep term? a quick google didn't turn up anything for me

Reply:
Outpoint is a Bitcoin term, pulled from the C++ code in bitcoind I believe. You can do a find at https://developer.bitcoin.org/glossary.html mentions it, for example. It's defined as exactly the above, and is the way to reference previous output(s) for funding a transaction.